src/AppContext.jsx

```diff
@@ -0,0 +1,388 @@
+      setStopwatchIsPaused(savedState.stopwatchIsPaused);
+
+      setCountdownTime(savedState.countdownTime);
+      setCountdownValue(savedState.countdownValue);
+      setCountdownIsRunning(savedState.countdownIsRunning);
+      setCountdownIsPaused(savedState.countdownIsPaused);
+
+      setXyTime(savedState.xyTime);
+      setXyTimeValue(savedState.xyTimeValue);
+      setXyCurrentRound(savedState.xyCurrentRound);
+      setXyIsRunning(savedState.xyIsRunning);
+      setXyIsPaused(savedState.xyIsPaused);
+
+      setTabataCountdown(savedState.tabataCountdown);
+      setTabataRestdown(savedState.tabataRestdown);
+      setTabataTotalCountdown(savedState.tabataTotalCountdown);
+      setTabataTotalRestdown(savedState.tabataTotalRestdown);
+      setTabataRounds(savedState.tabataRounds);
+      setTabataCurrentRound(savedState.tabataCurrentRound);
+      setTabataIsRunning(savedState.tabataIsRunning);
+      setTabataIsPaused(savedState.tabataIsPaused);
+    }
+  }, []);
+
+  const getTimerComponent = (title, index) => {
+    const components = {
+      Stopwatch: <Stopwatch controls={true} index={index} />,
+      Countdown: <Countdown controls={true} index={index} />,
+      XY: <XY controls={true} index={index} />,
+      Tabata: <Tabata controls={true} index={index} />,
+    };
+    return components[title];
+  };
+
+  // Stopwatch states
+  const [stopwatchTime, setStopwatchTime] = useState(0);
+  const [stopwatchTargetValue, setStopwatchTargetValue] = useState(0);
+  const [stopwatchIsRunning, setStopwatchIsRunning] = useState(false);
+  const [stopwatchIsPaused, setStopwatchIsPaused] = useState(false);
+
+  // Countdown states
+  const [countdownTime, setCountdownTime] = useState(0);
+  const [countdownValue, setCountdownValue] = useState(0);
+  const [countdownIsRunning, setCountdownIsRunning] = useState(false);
+  const [countdownIsPaused, setCountdownIsPaused] = useState(false);
+
+  // XY states
+  const [xyTime, setXyTime] = useState(0);
+  const [xyTimeValue, setXyTimeValue] = useState(0);
+  const [xyRounds, setXyRounds] = useState(2);
+  const [xyCurrentRound, setXyCurrentRound] = useState(1);
+  const [xyIsRunning, setXyIsRunning] = useState(false);
+  const [xyIsPaused, setXyIsPaused] = useState(false);
+
+  // Tabata states
+  const [tabataCountdown, setTabataCountdown] = useState(0);
+  const [tabataRestdown, setTabataRestdown] = useState(0);
+  const [tabataTotalCountdown, setTabataTotalCountdown] = useState(0);
+  const [tabataTotalRestdown, setTabataTotalRestdown] = useState(0);
+  const [tabataRounds, setTabataRounds] = useState(0);
+  const [tabataCurrentRound, setTabataCurrentRound] = useState(1);
+  const [tabataIsRunning, setTabataIsRunning] = useState(false);
+  const [tabataIsPaused, setTabataIsPaused] = useState(false);
+
+  // Save timer state when any relevant state changes
+  useEffect(() => {
+    if (stopwatchIsRunning || countdownIsRunning || xyIsRunning || tabataIsRunning) {
```

Comment:
You may also want to save the state when you add a new timer to the queue, even if no timer is running